internal/controller/issues.go

```diff
@@ -0,0 +1,171 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v56/github"
+	issues "github.com/matanamar10/github-issue-operator-hhome-assignment/api/v1alpha1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"time"
+)
+
+// searchForIssue checks if GithubIssue CRD has an issue in the repo
+func searchForIssue(issue *issues.GithubIssue, gitHubIssues []*github.Issue) *github.Issue {
+	for _, ghIssue := range gitHubIssues {
+		if ghIssue != nil && strings.EqualFold(*ghIssue.Title, issue.Spec.Title) {
+			return ghIssue
+		}
+	}
+	return nil
+}
+
+// UpdateIssueStatus updates the status of the GithubIssue CRD
+func (r *GithubIssueReconciler) UpdateIssueStatus(ctx context.Context, issue *issues.GithubIssue, githubIssue *github.Issue) error {
+	PRChange := r.CheckForPR(githubIssue, issue)
+	OpenChange := r.CheckIfOpen(githubIssue, issue)
+
+	if PRChange || OpenChange {
+		r.Log.Info("Updating Issue status", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+
+		if err := r.Client.Status().Update(ctx, issue); err != nil {
+			r.Log.Warn("Status update failed, attempting fallback", zap.Error(err))
+
+			if fallbackErr := r.Client.Update(ctx, issue); fallbackErr != nil {
+				r.Recorder.Event(issue, corev1.EventTypeWarning, "StatusUpdateFailed", fmt.Sprintf("Failed to update status: %v", fallbackErr))
+				return fmt.Errorf("failed to update status: %v", fallbackErr)
+			}
```

Comment:
What's this pattern? Don't think I've seen it... why are just attempting to update the entire object if the status update has failed? If the status update failed the it will be requeued again and in the next reconciliation loop it may succeed or fail